api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -249,6 +251,11 @@ type FelixConfigurationSpec struct {
 	HealthEnabled *bool   `json:"healthEnabled,omitempty"`
 	HealthHost    *string `json:"healthHost,omitempty"`
 	HealthPort    *int    `json:"healthPort,omitempty"`
+	// HealthTimeoutOverrides allows the internal watchdog timeouts of individual subcomponents to be
+	// overriden; example: "internal-dataplane-main-loop=30s,calculation-graph=2m".  This is useful for
+	// working around "false positive" liveness timeouts that can occur in particularly stressful workloads
+	// or if CPU is constrained.  For a list of active subcomponents, see Felix's logs.
+	HealthTimeoutOverrides string `json:"healthTimeoutOverrides,omitempty" validate:"omitempty,keyDurationList"`
```

Comment:
How feasible is it to make this a structured field?

For example:

```yaml
- name: InternalDataplaneMainLoop
  value: 30s
```
or

```
timeoutOverrides:
  internalDataplaneMainLoop: 30s
```

Or similar? I think these comma-separated string lists are fairly awkward to use. Their main benefit IIRC is that they are easier to encode into an env var, but we're pushing away from that being the primary means of configuring Felix.

I understand this might be a challenge given Felix's config merging machinery, but figured I'd ask!

Reply:
Done, made it a slice